File: tests/fixtures/vulnerable/csrf_no_token.ts
// Vulnerable: cookie-backed POST without CSRF protection
import express from "express";
import session from "express-session";

const app = express();
app.use(session({ secret: "key", cookie: { secure: true } }));

app.post("/api/transfer", (req, res) => {
  // State-changing with cookie session, no CSRF
  const { amount, to } = req.body;
  doTransfer(req.session.userId, to, amount);
  res.json({ status: "ok" });
});

File: tests/fixtures/vulnerable/csrf_no_token.py
"""Vulnerable: cookie-auth POST without CSRF protection."""
from flask import Flask, request, session

app = Flask(__name__)


@app.route("/transfer", methods=["POST"])
def transfer():
    """Bad: state-changing endpoint with cookie session, no CSRF."""
    amount = request.form["amount"]
    to_user = request.form["to"]
    do_transfer(session["user_id"], to_user, amount)
    return {"status": "ok"}
